lib/process-services-cloud/src/lib/task/task-filters/components/task-assignment-filter/task-assignment-filter.component.spec.ts

```diff
@@ -54,34 +70,135 @@ describe('EditTaskFilterCloudComponent', () => {
+
+        const candidateGroups = fixture.debugElement.query(By.css('[data-automation-id="adf-group-cloud-candidate-groups-filter"]'));
+        expect(component.candidateGroups.length).toEqual(0);
         expect(candidateGroups).toBeTruthy();
     });
 
+    it('should show the ASSIGNED_TO input', () => {
+        selectAssignmentType(AssignmentType.ASSIGNED_TO);
+
+        const candidateGroups = fixture.debugElement.query(By.css('[data-automation-id="adf-group-cloud-assigned-to-filter"]'));
+        expect(component.assignedUsers.length).toEqual(0);
+        expect(candidateGroups).toBeTruthy();
+    });
+
+    describe('status input change', () => {
+        it('should CREATED status set assignment type to UNASSIGNED', () => {
+            const createdStatusChange = new SimpleChange(null, 'CREATED', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.UNASSIGNED);
+        });
+
+        it('should ASSIGNED status set assignment type to ASSIGNED_TO', () => {
+            const createdStatusChange = new SimpleChange(null, 'ASSIGNED', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.ASSIGNED_TO);
+        });
+
+        it('should empty status set assignment type to NONE', () => {
+            const createdStatusChange = new SimpleChange(null, '', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.NONE);
+        });
+    });
+
+    describe('set initial assignment type', () => {
+        it('should set assignment type to ASSIGNED_TO if initial assignedUsers exists', () => {
+            component.taskFilterProperty = {
+                key: 'assignment',
+                label: 'mock-filter',
+                value: { assignedUsers: mockIdentityUsers },
+                type: 'assignment',
+                attributes: { assignedUsers: 'assignedUsers', candidateGroups: 'candidateGroups'}
+            };
+            component.ngOnInit();
+
+            expect(component.assignmentType).toEqual(AssignmentType.ASSIGNED_TO);
+        });
+
+        it('should set assignment type to CANDIDATE_GROUPS if initial candidateGroups exists', () => {
+            component.taskFilterProperty = {
+                key: 'assignment',
+                label: 'mock-filter',
+                value: { candidateGroups: mockIdentityGroups },
+                type: 'assignment',
+                attributes: { assignedUsers: 'assignedUsers', candidateGroups: 'candidateGroups'}
+            };
+            component.ngOnInit();
+
+            expect(component.assignmentType).toEqual(AssignmentType.CANDIDATE_GROUPS);
+        });
+
+        it('should set assignment type to CANDIDATE_GROUPS if initial candidateGroups and assignedUsers exists', () => {
```

Comment:
is that possible from the UI to have both candidate groups and assigned users in this filter?

Reply:
no, unit test removed